leshan-core/src/main/java/org/eclipse/leshan/core/request/WriteRequest.java

```diff
@@ -29,23 +36,116 @@
  */
 public class WriteRequest extends AbstractDownlinkRequest<WriteResponse> {
 
+    public enum Mode {
+        REPLACE, UPDATE
+    }
+
     private final LwM2mNode node;
     private final ContentFormat contentFormat;
+    private final Mode mode;
+
+    // ***************** write object ****************** //
+
+    public WriteRequest(final Mode mode, final ContentFormat contentFormat, final int objectId,
```

Comment:
when I say "mutable", I mean "mutable just for the content type to be set before marshalling" :)